### DISCUSSION TITLE
How to save a checkpoint every n steps and overwrite the previous saved checkpoints?

### DISCUSSION THREAD

**Discussion by stanleyshly** [Lightning-AI/pytorch-lightning#6329] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
How do I save checkpoints every, as well as deleting and/or overwriting the previously saved checkpoints?

**Comment by SkafteNicki** [comment, marked as answer]
Just to be sure, do you mean every n step or every n epoch.
For doing it every n epoch, you can initialize model checkpoint with `period` parameter
``` python
checkpoint = pl.callbacks.ModelCheckpoint(period=n)
```

**Reply by SkafteNicki** [reply]
Update, when this PR https://github.com/PyTorchLightning/pytorch-lightning/pull/6146 gets merged it will be possible to save checkpoint every n training step

**Reply by stanleyshly** [reply]
Thank you for the response, I'll be eagerly awaiting the next version
